### PR TITLE
*: Protocol Buffer docs auto-generate script

### PR DESCRIPTION
**Pull request by gyuho**
**It's experimental**. One can run './scripts/genproto.sh -g'

Fix https://github.com/coreos/etcd/issues/5100.

### CODE REVIEW

**Comment by gyuho** [comment]
Wrote a simple CLI https://github.com/coreos/prodoc.

Considered https://github.com/estan/protoc-gen-doc but this doesn't support any proto files with `import "gogoproto/gogo.proto";`.

If I have more time, I will replace the parser with actual golang parser https://github.com/golang/protobuf.

**Comment by xiang90** [comment]
Can you show us the generate markdown output?

**Comment by gyuho** [comment]
@xiang90 Here's the sample markdown output https://github.com/gyuho/etcd/blob/proto_gen/Documentation/api_etcdserverpb.md.

**Comment by gyuho** [comment]
@xiang90 Fixed up. Changed the order and print `Emtpy ...` for 0 method. https://github.com/gyuho/etcd/blob/proto_gen/Documentation/api_etcdserverpb.md. PTAL.

**Comment by gyuho** [comment]
@xiang90 @heyitsanthony Just updated it https://github.com/gyuho/etcd/blob/proto_gen/Documentation/api_protocol_buffer.md.

1. All docs in one markdown
2. only expose public message (not including `InternalRaftMessage`, etc)

PTAL.

Thanks.

**Comment by gyuho** [comment]
hold on... lease.proto missing.

**Comment by gyuho** [comment]
Ok now ready for review.

https://github.com/gyuho/etcd/blob/proto_gen/Documentation/api_protocol_buffer.md

PTAL.

**Comment by xiang90** [comment]
LGTM. Defer to @heyitsanthony

File: Documentation/api_protocol_buffer.md [hunk 1]
```diff
@@ -0,0 +1,771 @@
+### Protocol Buffer API
```
Comment (xiang90) [inline Documentation/api_protocol_buffer.md]: api_reference_v3.md

**Comment by heyitsanthony** [comment]
lgtm

**Comment by gyuho** [comment]
@xiang90 Changed file name to `api_reference_v3.md`. Merging in greens. Thanks.

**Pull request merged**